tests/test-integrity.js

```diff
@@ -0,0 +1,110 @@
+var assert = require("assert");
+var fs = require('fs');
+
+var Data = require("./data");
+var moment = require("moment");
+var utils = require("./utils");
+
+
+describe("All data", function() {
+  describe("in sources.js", function() {
+    describe("that have 'release_date' value", function() {
+      it("should have an 'announcement_date' that comes before it", function() {
+        var errors = [];
+        var source;
+
+        for (var i = 0; i < Data.sources.length; i++) {
+          source = Data.sources[i];
+
+          if (source.release_date && source.announcement_date) {
+            if (!moment(source.announcement_date).isBefore(moment(source.release_date))) {
+              errors.push(
+                utils.buildDataHeader(Data, 'sources', i) +
+                ": 'release_date' comes before 'announcement_date'."
+              )
+            }
+          }
+        }
+        utils.outputAllErrors(errors);
+      });
+    });
+
+    describe("that have 'release_date' value in the past", function() {
+      it("should have 'released' value set to true", function() {
+        var errors = [];
+        var source;
+        var now = moment();
+
+        for (var i = 0; i < Data.sources.length; i++) {
+          source = Data.sources[i];
+
+          if (source.release_date && now.isAfter(moment(source.release_date))) {
+
+            if (typeof source.released === "undefined") {
+              errors.push(
+                utils.buildDataHeader(Data, 'sources', i) +
+                ": 'released' field should be true, but it's undefined."
+              )
+            } else if (!source.released) {
+              errors.push(
+                utils.buildDataHeader(Data, 'sources', i) +
+                ": 'released' field  should be true, but it's set to false."
+              )
```

Comment:
This whole `if/else` block can be replaced by just 
```js
errors.push(
  utils.buildDataHeader(Data, 'sources', i) +
  ": 'released' field should be true, but it's set to " + source.released + "."
)
```
and `source.released` will be stringified to either `false` or `undefined`.